sdk/keyvault/Azure.Security.KeyVault.Keys/src/JsonWebKey.cs

```diff
@@ -120,6 +120,250 @@ public JsonWebKey(string keyId, string keyType, IList<string> keyOperations)
+                {
+                    KeyOps.Add(KeyOperationsExtensions.ParseFromString(element.Value.ToString()));
+                }
+            }
+            if (json.TryGetProperty("curveName", out JsonElement curveName))
+            {
+                CurveName = curveName.GetString();
+            }
+            if (json.TryGetProperty("n", out JsonElement n))
+            {
+                N = Encoding.ASCII.GetBytes(n.GetString());
+            }
+            if (json.TryGetProperty("e", out JsonElement e))
+            {
+                E = Encoding.ASCII.GetBytes(e.GetString());
+            }
+            if (json.TryGetProperty("dp", out JsonElement dp))
+            {
+                DP = Encoding.ASCII.GetBytes(dp.GetString());
+            }
+            if (json.TryGetProperty("dq", out JsonElement dq))
+            {
+                DQ = Encoding.ASCII.GetBytes(dq.GetString());
+            }
+            if (json.TryGetProperty("qi", out JsonElement qi))
+            {
+                QI = Encoding.ASCII.GetBytes(qi.GetString());
+            }
+            if (json.TryGetProperty("p", out JsonElement p))
+            {
+                P = Encoding.ASCII.GetBytes(p.GetString());
+            }
+            if (json.TryGetProperty("q", out JsonElement q))
+            {
+                Q = Encoding.ASCII.GetBytes(q.GetString());
+            }
+            if (json.TryGetProperty("x", out JsonElement x))
+            {
+                X = Encoding.ASCII.GetBytes(x.GetString());
+            }
+            if (json.TryGetProperty("y", out JsonElement y))
+            {
+                Y = Encoding.ASCII.GetBytes(y.GetString());
+            }
+            if (json.TryGetProperty("d", out JsonElement d))
+            {
+                D = Encoding.ASCII.GetBytes(d.GetString());
+            }
+            if (json.TryGetProperty("k", out JsonElement k))
+            {
+                K = Encoding.ASCII.GetBytes(k.GetString());
+            }
+            if (json.TryGetProperty("t", out JsonElement t))
+            {
+                T = Encoding.ASCII.GetBytes(t.GetString());
+            }
+        }
+
+        internal override void WriteProperties(ref Utf8JsonWriter json)
+        {
+            if (KeyType != default)
+            {
+                json.WriteString("kty", KeyTypeExtensions.ParseToString(KeyType));
+            }
+            if (KeyOps != null)
+            {
+                json.WriteStartArray("key_ops");
+                foreach (var operation in KeyOps)
+                {
+                    json.WriteStringValue(KeyOperationsExtensions.ParseToString(operation));
+                }
+                json.WriteEndArray();
+            }
+            if (!string.IsNullOrEmpty(CurveName))
+            {
+                json.WriteString("crv", CurveName);
+            }
+            if (N != null)
+            {
+                json.WriteString("n", Encoding.ASCII.GetString(N));
```

Comment:
You can write ASCII bytes directly using WriteStringValue, i.e. no need to encode to System.String first.

cc: @ahsonkhan.

Reply:
Looking more closely, it looks like the goal is to write a base-64 decoded byte[] as a JSON string. The `WriteBase64String` API should be used for that (with the actual byte[] passed in).

Reply:
I see that this new functionality was added 4 days ago with commit https://github.com/dotnet/corefx/commit/82408cd90f4d4573d502e8df2ca437b35e6a37f7
Do you know when these changes are going to hit the NuGet package?

Reply:
The usual timeline would be when preview6 ships (~2 weeks). Alternatively, you could depend on the nightly build Nuget feed and grab the latest package from that:
```text
https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json
```

Reply:
Again these fields are base64url encoded not base64 encoded.  @ashonkhan would it be possible to add base64url support in Utf8JsonWriter?

---
In reply to: [290105191](https://github.com/Azure/azure-sdk-for-net/pull/6457#discussion_r290105191) [](ancestors = 290105191)

Reply:
> would it be possible to add base64url support in Utf8JsonWriter?

We don't have base64url support built-in within .NET Core (which the json support would be built on top), so not for 3.0.

Reply:
> Again these fields are base64url encoded not base64 encoded

Thanks for clarifying, @schaabs. Are their tests to validate the properties are set to the base64 url 
 bytes and serialize as base64 url encoded text? The tests should have caught this, no?

Reply:
Yes. Tests will be added next week